nncf/torch/nncf_network.py

```diff
@@ -715,10 +700,255 @@ def get_node_to_op_address_mapping(self) -> Dict[NNCFNodeName, OperationAddress]
+        a "denylist". If left unspecified, nothing will be ignored.
+        :param target_scopes: A list of scopes in the model for which NNCF handling should be applied. Functions as
+        an "allowlist". If left unspecified, everything will be targeted.
+        :param wrap_outputs_fn: Same as `wrap_inputs_fn`, but for marking model outputs with
+        `nncf.torch.dynamic_graph.io_handling.nncf_model_output` calls.
+        :return: The same object as passed in `original_model`, but with internal modules extended/replaced for
+        purposes of further NNCF compression, and its class dynamically extended with the `NNCFNetwork` as a base class.
+        The object will pass both isinstance(retval, original_model.__class__) and isinstance(retval, NNCFNetwork)
+        checks.
+        """
+        original_class = original_model.__class__
+        original_model._nncf = NNCFNetworkInterface(original_model,
+                                                    input_infos,
+                                                    dummy_forward_fn,
+                                                    wrap_inputs_fn,
+                                                    scopes_without_shape_matching,
+                                                    ignored_scopes,
+                                                    target_scopes,
+                                                    wrap_outputs_fn)  # pylint:disable=protected-access
+        # The new class will also have an adjusted metaclass to avoid a "metaclass conflict" upon
+        # class creation
+        original_metaclass = type(original_model.__class__)
+        class_creation_kwds = {}
+        if original_metaclass is not type:
+            new_metaclass = types.new_class(name=original_metaclass.__name__, bases=(NNCFNetworkMeta,
+                                                                                     original_metaclass))
+            class_creation_kwds["metaclass"] = new_metaclass
+        new_class = types.new_class(name=original_model.__class__.__name__,
+                                    bases=(NNCFNetwork, original_model.__class__),
+                                    kwds=class_creation_kwds)
+        # Make the signature of the forward on the resulting object same as for
+        # the original forward.
+        fn = NNCFNetwork.forward
+        new_forward = types.FunctionType(fn.__code__, fn.__globals__, fn.__name__,
+                                         fn.__defaults__, fn.__closure__)
+        new_forward.__dict__.update(fn.__dict__)
+        new_forward.__signature__ = inspect.signature(original_class.forward)
+        new_class.forward = new_forward
+
+        # Make resulting class keep __module__ attributes of the original class,
+        # otherwise these will point to NNCF
+        new_class.__module__ = original_class.__module__
+        original_model.__class__ = new_class
+
+        if isinstance(original_model, torch.nn.Sequential):
```

Comment:
Do we have such model in test scope? if not, makes sense to check it.

Reply:
`tests.torch.test_nncf_network.test_can_work_with_sequential_models` is added in this PR already